### PR TITLE
Bug/pla 2982 strip hints

### PR DESCRIPTION
**Pull request by crleblanc**
# Citrine Python PR

## Description 
This PR adds a 'build_py' custom builder, wrapping the existing run method with code that strips all of the type hints that are incompatible with Python 3.5 to enable support for Python 3.5+ with a single wheel package.  It also uses an older print syntax instead of the newer f-string.

I have replaced the strip_hints.sh script with a Python function in setup.py.  The script had some issues, if strip-hints was not installed it would wipe all of your source code and would not report an error.  Strip_hints is available as a Python module so it's easy to use from setup.py which will report an error if it's not installed.

Manual testing shows the type hints are removed in the wheel file and it imports cleanly in Python 3.5 (via Docker).  I've also run "python setup.py test" after pip-installing from requirements.txt and test_requirements.txt and all 5 tests pass.

I've updated the strip-hints dependency.  I can't find a changelog or release for this version, but a higher number must be better, right?  It produces the expected results so I think it's a safe change.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,8 +1,30 @@
...
+
+STRIP_DIRS = ["src", "tests"]
+
+
+def strip_file(file_path):
+    from strip_hints import strip_file_to_string
+
+    if file_path.endswith('.py'):
+        stripped = strip_file_to_string(file_path, to_empty=True, only_assigns_and_defs=True)
+        print(file_path)
```
Comment (maxhutch) [inline setup.py]: Is this a debugging print?

File: setup.py [hunk 2]
```diff
@@ -1,8 +1,30 @@
...
+        with open(file_path, 'w') as file_obj:
+            file_obj.write(stripped)
+
+
+def strip_all_hints(dirnames):
+    print('stripping type annotations from: {}'.format(dirnames))
```
Comment (maxhutch) [inline setup.py]: Is this a debugging print?

File: setup.py [hunk 3]
```diff
@@ -39,14 +68,15 @@ def run(self):
           "requests>=2.22.0,<3",
           "pyjwt>=1.7.1,<2",
           "arrow>=0.15.4,<0.16",
-          "strip-hints>=0.1.5,<=0.1.7",
+          "strip-hints>=0.1.8",
```
Comment (maxhutch) [inline setup.py]: Can we keep an upper bound here?

File: setup.py (hunk 1 above)
Comment (crleblanc) [reply setup.py]: I was implementing what the strip_hints.sh shell script had.  It prints out each file it processes.  I don't have a strong opinion on them so I'm happy to remove them.

File: setup.py (hunk 2 above)
Comment (crleblanc) [reply setup.py]: Same as above, it's doing the same operation as the shell script.

File: setup.py (hunk 3 above)
Comment (crleblanc) [reply setup.py]: Sure, or we could just lock it to 0.1.8.

File: setup.py (hunk 1 above)
Comment (bfolie) [reply setup.py]: The print statements were originally in there just to ensure the actions were being logged in the Travis output. I don't think it hurts to be verbose in this case 🤷‍♂

Comment (crleblanc) [reply setup.py]: Usually I'm not a fan of too many print statements but in this case I think it's useful.  There aren't very many files and it's nice to see that it's running as expected.

**Pull request merged**